samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -39,7 +40,8 @@ public AbstractApplicationRunner(Config config) {
   @Override
   public StreamSpec streamFromConfig(String streamId) {
     StreamConfig streamConfig = new StreamConfig(config);
-    String physicalName = streamConfig.getPhysicalName(streamId, streamId);
+    String defaultPhysicalName = String.format("%s-%s-%s", config.get(JobConfig.JOB_NAME()), config.get(JobConfig.JOB_ID(), "1"), streamId);
```

Comment:
Question again on the configuration scope: are we "borrowing" job.name and job.id as app level configuration here? I think that the unique intermediate StreamSpec should be at app level?